From 6c34643479c9060ab0c30105e2b384fcf3277c34 Mon Sep 17 00:00:00 2001
From: Albin B <albin.bala.krishnan@intel.com>
Date: Wed, 23 Sep 2015 15:41:32 +0530
Subject: [PATCH] [EM/GPIO]: Process all the extcon event for vbus
 connect/disconnect

When plug/unplug iterations, TI device resolves as SNK and FFD as SRC,
however FFD never provides VBUS and communication fails. TI device is
a DRP so it randomly resolved as either DFP or UFP, this happens
during a transition from DFP to UFP. So process all the extcon events
for vbus connect/disconnect.

Change-Id: I42943aa75e0e58b7afc96602098247293fb14685
Tracked-On: https://jira01.devtools.intel.com/browse/GMINL-18896
Signed-off-by: Albin B <albin.bala.krishnan@intel.com>
---
 drivers/misc/intel_wcove_gpio.c | 9 +--------
 1 file changed, 1 insertion(+), 8 deletions(-)

diff --git a/drivers/misc/intel_wcove_gpio.c b/drivers/misc/intel_wcove_gpio.c
index 65b7fbd87775..e30b3cbb16ab 100644
--- a/drivers/misc/intel_wcove_gpio.c
+++ b/drivers/misc/intel_wcove_gpio.c
@@ -54,7 +54,6 @@ struct wcove_gpio_info {
 	struct list_head gpio_queue;
 	struct work_struct gpio_work;
 	struct mutex lock;
-	bool is_vbus_connected;
 	spinlock_t gpio_queue_lock;
 };
 
@@ -141,14 +140,13 @@ static void wcgpio_ctrl_worker(struct work_struct *work)
 		dev_info(&info->pdev->dev,
 				"%s:%d state=%d\n", __FILE__, __LINE__,
 				evt->is_src_connected);
-		info->is_vbus_connected = evt->is_src_connected;
 		retry_count = 0;
 
 		do {
 			/* loop is to update the vbus state in case fails, try
 			 * again upto max retry count */
 			ret = wcgpio_update_vbus_state(info,
-					info->is_vbus_connected);
+					evt->is_src_connected);
 			if (ret != -EAGAIN)
 				break;
 
@@ -191,11 +189,6 @@ static int wcgpio_check_events(struct wcove_gpio_info *info,
 	}
 
 	evt->is_src_connected = extcon_get_cable_state(edev, "USB_TYPEC_SRC");
-	if (evt->is_src_connected == info->is_vbus_connected) {
-		dev_info(&info->pdev->dev, "already source %s\n",
-			evt->is_src_connected ? "Connected" : "Disconnected");
-		return 0;
-	}
 	dev_info(&info->pdev->dev,
 			"[extcon notification] evt: Provider - %s\n",
 			evt->is_src_connected ? "Connected" : "Disconnected");
-- 
2.6.0.rc0.2.g7662973

